spec/lib/http_checker_spec.rb

```diff
@@ -0,0 +1,11 @@
+require "rails_helper"
+
+RSpec.describe LinkChecker::UriChecker::HttpChecker do
+  let(:uri) { URI("http://example.invalid/") }
+  subject { described_class.new(uri) }
+  it "should not throw an exception if the response is nil" do
+    stub_request(:get, "http://example.invalid/").to_return(status: 500, body: "", headers: {})
```

Comment:
We can DRY this up - and I also don't think we necessarily need an empty body or headers 🤔 

```suggestion
    stub_request(:get, uri).to_return(status: 500)
```

I also wonder if this `stub_request` is actually doing anything - I imagine the `allow(subject).to receive` on the next line is setting up the test how we want. What happens if we remove `stub_request`?

Reply:
Ah, I added the `stub_request` because of an error but that seems like it was a red herring — presumably I'd gotten the other stub wrong at that stage.